src/main/scala/outwatch/dom/DomTypes.scala

```diff
@@ -0,0 +1,139 @@
+      VTree(tagName, Seq.empty)
+  }
+
+  object CodecBuilder {
+    type Attribute[T, _] = AttributeBuilder[T]
+    type Property[T, _] = PropertyBuilder[T]
+
+    def encode[V](codec: Codec[V, _]): V => Attribute.Value = codec match {
+      //The BooleanAsAttrPresenceCodec does not play well with snabbdom. it
+      //encodes true as "" and false as null, whereas snabbdom needs true/false
+      //of type boolean (not string) for toggling the presence of the attribute.
+      case _: BooleanAsAttrPresenceCodec.type => identity
+      case _: BooleanAsIsCodec.type => identity
+      // codec.encode may encode to null values
+      case _ => (codec.encode _) andThen (v => if (v != null) v.toString else null)
+    }
+  }
+
+  abstract class ObservableEventPropBuilder(target: dom.EventTarget) extends EventPropBuilder[Observable, dom.Event] {
+    override def eventProp[V <: dom.Event](key: String): Observable[V] = Observable.create { obs =>
+      target.addEventListener(key, obs.next _)
+    }
+  }
+
+  trait SimpleStyleBuilder extends StyleBuilders[IO[Style]] {
+
+    implicit def StyleIsBuilder[T](style: keys.Style[T]): StyleBuilder[T] = new StyleBuilder[T](style.cssName)
+
+    override protected def buildDoubleStyleSetter(style: keys.Style[Double], value: Double): IO[Style] = style := value
+    override protected def buildIntStyleSetter(style: keys.Style[Int],value: Int): IO[Style] = style := value
+    override protected def buildStringStyleSetter(style: keys.Style[_],value: String): IO[Style] = new StyleBuilder[Any](style.cssName) := value
+  }
+}
+import DomTypesBuilder._
+
+trait DomTypes
+  extends DomTags
+  with DomAttrs
+  with DomReflectedAttrs
+  with DomProps
+  with DomEvents
+  with DomStyles
+
+trait DomTags
+  extends GroupingTags[GenericVNode, VNode_]
+  with TextTags[GenericVNode, VNode_]
+  with FormTags[GenericVNode, VNode_]
+  with SectionTags[GenericVNode, VNode_]
+  with TableTags[GenericVNode, VNode_]
+  with VNodeBuilder
+object DomTags extends DomTags
+
+trait DomTagsExtra
+  extends DocumentTags[GenericVNode, VNode_]
+  with EmbedTags[GenericVNode, VNode_]
+  with MiscTags[GenericVNode, VNode_]
+  with VNodeBuilder
+object DomTagsExtra extends DomTagsExtra
+
+trait DomAttrs
+  extends Attrs[AttributeBuilder]
+  with AttrBuilder[AttributeBuilder] {
+
+  override protected def attr[V](key: String, codec: Codec[V, String]): AttributeBuilder[V] =
+    new AttributeBuilder(key, codec.encode _)
+}
+object DomAttrs extends DomAttrs
+
+trait DomReflectedAttrs
+  extends ReflectedAttrs[CodecBuilder.Attribute]
+  with DomAttrsCompat[CodecBuilder.Attribute]
+  with ReflectedAttrBuilder[CodecBuilder.Attribute] {
+
+  override protected def reflectedAttr[V, DomPropV](
+    attrKey: String,
+    propKey: String,
+    attrCodec: Codec[V, String],
+    propCodec: Codec[V, DomPropV]
+  ): AttributeBuilder[V] =
+    new AttributeBuilder(attrKey, CodecBuilder.encode(attrCodec))
```

Comment:
Here, we can decide whether we want to represent the reflected attributes as attributes or properties. See https://github.com/raquo/scala-dom-types#reflected-attributes

Reply:
Yes, I see. I think properties...

Reply:
I have to admit, I do not really know what disadvantages or advantages there might be in practice.

Reply:
Before we had everything as attributes, but I tend lean towards properties as well.

Reply:
It shouldn't really matter for reflected attributes (because they are reflected). Using properties is somewhat safer because they would not accept invalid values.